resolver/map.go

```diff
@@ -136,3 +136,134 @@ func (a *AddressMap) Values() []any {
 	}
 	return ret
 }
+
+type endpointNodeList []endpointNode
+
+type endpointNode struct {
+	addrs map[string]int
+	val   any
+}
+
+// Equal returns whether the unordered set of addrs counts are the same between
+// the endpoint nodes.
```

Comment:
Why use the counts?  I don't think an endpoint should include the same address multiple times, and if it does, it seems like we'd just ignore the duplicates.

Reply:
I use the counts because it feels like a defensive way with no cost incured. I.e. I either do a presence check with a struct{} or bool or I have the value be an int. I know it's not an intended use case of an endpoint to have the same address (by string) multiple times but I added this since it seems to be similar to add it vs. presence.

Reply:
Ping. Let's make this a set, which is how the docstring below indicates it behaves as well.

Reply:
This is still technically a set, right? I don't see any downside to doing it this way, and I think it is more technically correct to take into account address counts (still unordered). Technically, what do you see as the pro of AABC == ABC vs AABC != ABC? I feel like it makes more sense that they're not equivalent (and this case will never happen in practice anyway). It allows the resolver to encode more information in the endpoint (250 ms delay fail address a, then try address a again)...hmmm. I don't think happy eyeballs makes sense wrt doing aa, but right now unless we use keys() to pass down to child the duplicated addresses in an endpoint will be conveyed to the pick first child. So I think whether we account for address counts here is orthogonal to whether we want to pass down AABC to pick first child and have it do happy eyeballs algorithm on AA (relates to that discussion we had monday about pick first never reporting TF).

Reply:
I put this in the dualstack design chat. Will wait for more discussion before choosing to implement this suggestion or not.

Reply:
Mark said to strip counts, so will implement this suggestion.